gdc_client/upload/client.py

```diff
@@ -135,62 +135,79 @@ def __init__(self, token, processes, server, part_size,
 
         self.files = files
         self.incompleted = deque(copy.deepcopy(self.files))
+
         if not (server.startswith('http://') or server.startswith('https://')):
             server = 'https://' + server
+
         self.server = server
         self.multipart = multipart
         self.upload_id = None
         self.debug = debug
         self.processes = processes
         self.part_size = (max(part_size, MIN_PARTSIZE)/PAGESIZE+1)*PAGESIZE
-        self._metadata = None
+        self._metadata = {}
         self.resume_path = "resume_{}".format(self.manifest_name)
 
-    @property
-    def metadata(self):
-        return self._metadata or self.get_metadata(self.node_id)
+    def metadata(self, field):
+        return self._metadata.get(field) or self.get_metadata(self.node_id)[field]
```

Comment:
hmm, I think this is a _do what I mean to get global metadata or metadata about a file_? tbh, seems confusing given the potential namespace collision